crates/paimon/src/spec/snapshot.rs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/*!
+ * Snapshots
+ */
+
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use typed_builder::TypedBuilder;
+
+/// Type of changes in this snapshot.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/db8bcd7fdd9c2705435d2ab1d2341c52d1f67ee5/paimon-core/src/main/java/org/apache/paimon/Snapshot.java#L506>.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CommitKind {
+    /// Change flushed from the mem table.
+    Append,
+    /// Changes by compacting existing data files.
+    Compact,
+    /// Changes that clear up the whole partition and then add new records.
+    Overwrite,
+    /// Collect statistics.
+    Analyze,
+}
+
+/// Snapshot for paimon.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/db8bcd7fdd9c2705435d2ab1d2341c52d1f67ee5/paimon-core/src/main/java/org/apache/paimon/Snapshot.java#L68>.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, TypedBuilder)]
+pub struct Snapshot {
+    /// version of snapshot
+    version: i32,
+    id: i64,
+    schema_id: i64,
+    /// a manifest list recording all changes from the previous snapshots
+    base_manifest_list: String,
+    /// a manifest list recording all new changes occurred in this snapshot
+    delta_manifest_list: String,
+    /// a manifest list recording all changelog produced in this snapshot
+    #[builder(default = None)]
+    change_log_manifest_list: Option<String>,
+    /// a manifest recording all index files of this table
+    #[builder(default = None)]
+    index_manifest: Option<String>,
+    commit_user: String,
+    /// Mainly for snapshot deduplication.
+    ///
+    /// If multiple snapshots have the same commitIdentifier, reading from any of these snapshots
+    /// must produce the same table.
+    ///
+    /// If snapshot A has a smaller commitIdentifier than snapshot B, then snapshot A must be
+    /// committed before snapshot B, and thus snapshot A must contain older records than snapshot B.
+    commit_identifier: i64,
+    commit_kind: CommitKind,
+    time_millis: i64,
+    log_offsets: HashMap<i32, i64>,
```

Comment:
I think we don't need to add this field.
We don't write it, and we don't have requirement to read it.

Reply:
> I think we don't need to add this field. We don't write it, and we don't have requirement to read it.

OK, I will remove it later.